ng-dev/pr/merge/target-label.ts

```diff
@@ -6,9 +6,34 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {MergeConfig, TargetLabel} from './config';
-import {getTargetLabels} from './defaults/labels';
+import {MergeConfig} from './config';
 import {matchesPattern} from './string-pattern';
+import {getTargetLabelsForActiveReleaseTrains} from './defaults';
+import {GithubConfig} from '../../utils/config';
+import {Commit} from '../../commit-message/parse';
+import {assertChangesAllowForTargetLabel} from './validations';
+import {PullRequestFailure} from './failures';
+
+/** Describes possible values that can be returned for `branches` of a target label. */
+export type TargetLabelBranchResult = string[] | Promise<string[]>;
+
+/**
+ * Matcher that can resolve a Github label to a list of branches into which a pull
+ * request should be merged into..
+ */
+export interface TargetLabel {
+  /** Pattern that matches the given target label. */
+  pattern: RegExp | string;
+  /**
+   * List of branches a pull request with this target label should be merged into.
+   * Can also be wrapped in a function that accepts the target branch specified in the
+   * Github Web UI. This is useful for supporting labels like `target: development-branch`.
+   *
+   * @throws {InvalidTargetLabelError} Invalid label has been applied to pull request.
+   * @throws {InvalidTargetBranchError} Invalid Github target branch has been selected.
+   */
+  branches: TargetLabelBranchResult | ((githubTargetBranch: string) => TargetLabelBranchResult);
```

Comment:
To make this simpler should we just change this to

```suggestion
  branches: (githubTargetBranch: string) => Promise<TargetLabelBranchResult>;
```

Since its no longer user configuration that defines `TargetLabel`. Currently only one of them is a non-function, and we always `await` the `branches` value anyway.  We could then drop the `getBranchesFromTargetLabel` function as well.

Reply:
👍